force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,128 @@
+            new Lead(FirstName='Test', LastName='Coverage', Company='UniqueBizzzzz' + System.currentTimeMillis(), Title='exec', Email='test2' + System.currentTimeMillis() + '@example2.com', Phone='9876543266', Industry='Retailer', LeadSource='Referralll'),
+            new Lead(FirstName='Testing', LastName='Coveraged', Company='UniqueBizzzxz' + System.currentTimeMillis(), Title='assist', Email='test21' + System.currentTimeMillis() + '@example1.com', Phone='9878543266', Industry='Retasiler', LeadSource='Refesrralll')
+
+
+        };
+        insert leads;
+
+        Contact contact = new Contact(FirstName='John', LastName='Doe', Email='unique_contact' + System.currentTimeMillis() + '@example.com');
+        insert contact;
+    }
+
+
+
+  
+    
+    @isTest
+    static void testHandleTitleNormalization() {
+        List<Lead> leads = [SELECT Id, Title FROM Lead];
+        Test.startTest();
+        LeadTriggerHandler.handleTitleNormalization(leads);
+        Test.stopTest();
+
+        for (Lead lead : leads) {
+            if (lead.Title == 'vp') {
+                System.assertEquals('Vice President', lead.Title, 'Title should be normalized to Vice President');
+            } else if (lead.Title.contains('mgr')) {
+                System.assertEquals('Manager', lead.Title, 'Title should be normalized to Manager');
+            } else if (lead.Title.contains('exec')) {
+                System.assertEquals('Executive', lead.Title, 'Title should be normalized to Executive');
+            } else if (lead.Title.contains('assist')) {
+                System.assertEquals('Assistant', lead.Title, 'Title should be normalized to Assistant');
+            }
+        }
+    }
+
+    @isTest
+    static void testHandleAutoLeadScoring() {
+        List<Lead> leads = [SELECT Id, LeadSource, Email, Phone, Industry, Lead_Score__c FROM Lead];
+        Test.startTest();
+        LeadTriggerHandler.handleAutoLeadScoring(leads);
+        Test.stopTest();
+
+        for (Lead lead : leads) {
+            if (lead.LeadSource == 'Website' && lead.Email != null) {
+                System.assert(lead.Lead_Score__c >= 3, 'Score should account for website source and email presence');
+            }
+            if (lead.Phone != null) {
+                System.assert(lead.Lead_Score__c >= 5, 'Score should account for phone presence');
+            }
+            if (lead.Industry == 'Technology') {
+                System.assert(lead.Lead_Score__c >= 10, 'Score should account for technology industry');
+            }
+            System.assert(lead.Lead_Score__c <= 18, 'Score should not exceed the maximum limit');
+        }
+    }
+
+@isTest
+static void testHandleLeadAutoConvert() {
+    delete [SELECT Id FROM Contact WHERE Email = 'convert@example.com'];
+    delete [SELECT Id FROM Lead WHERE Email = 'convert@example.com'];
```

Comment:
Instead of having to delete these leads to start the test, think about creating a method in a util class with a doInsert parameter and when you call the method, you decide if you want to insert or not.